plugins/compose/compiler-hosted/integration-tests/src/jvmTest/kotlin/androidx/compose/compiler/plugins/kotlin/FunctionBodySkippingTransformTests.kt

```diff
@@ -599,6 +599,23 @@ class FunctionBodySkippingTransformTests(
         """
     )
 
+    @Test
+    fun testInternalStableUnstableParams(): Unit = comparisonPropagation(
+        """
+            internal class Foo(var value: Int = 0)
+        """,
+        """
+            @Composable
+            internal fun CanSkip(foo: Foo = Foo()) {
```

Comment:
I am not sure this test makes a lot of sense as is.

You want to check for codegen of parameters with the types of `internal class FooStable(val value: Int)` and `internal class FooUnstable(var value: Int)` instead of unstable class only.

The other thing I noticed is that unstable class produced `changedInstance` rather than `changed` call, which tells me that `changedInstance` is always generated for `Runtime` stability. Please file a bug for this.

Also no need to check default arguments, that does not affect stability here.